### ISSUE TITLE
bug: float32 class-moment dead band — GEVD variance/skew cancel to ~0 for ξ ∈ (GUMBEL_THRESHOLD, ~3e-4)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/xtremax#88]
## Problem

Found by the epic-#41 scipy oracle tests (PR #87): for tiny nonzero ξ the class-layer exact moment formulas cancel catastrophically in float32.

`GeneralizedExtremeValueDistribution.variance` uses

```
(σ²/ξ²) · (Γ(1−2ξ) − Γ²(1−ξ))
```

For ξ = 1e-5 the two Γ terms are both `1 + O(1e-5)`; their true difference is `O(ξ²·π²/6) ≈ 1.6e-10`, far below float32 eps (1.2e-7), so the subtraction rounds to exactly 0 and the variance evaluates to **0 instead of σ²π²/6 ≈ 6.58** (total loss, not degraded accuracy). The Gumbel closed form takes over only below `GUMBEL_THRESHOLD = 1e-7`, leaving a dead band roughly ξ ∈ (1e-7, 3e-4) where variance/skew/kurtosis are garbage in float32. `skew()` and `kurtosis()` share the same Γ-difference structure.

## Reproduction

```python
import jax.numpy as jnp
from xtremax.distributions import GeneralizedExtremeValueDistribution
d = GeneralizedExtremeValueDistribution(1.5, 2.0, concentration=1e-5)
float(d.variance)   # 0.0 — true value ≈ 6.5799 (σ²π²/6 at this ξ)
```

## Expected Behavior

Moments accurate (or at worst smoothly interpolated) through the small-ξ band, matching `scipy.stats.genextreme(-ξ).stats()` within float32 tolerance.

## Suggested Fixes (either)

1. **Series expansion** for the Γ-difference terms in the band (e.g. `Γ(1−2ξ) − Γ²(1−ξ) = ξ²π²/6 + O(ξ³)` with the next-order γ/ζ(3) terms), switched at a dtype-aware threshold — mirrors how the log-prob/cdf smooth kernels already handle the band.
2. **Wider moment-specific threshold**: switch variance/skew/kurtosis to the Gumbel closed forms below ~`sqrt(eps)`-scale ξ (≈ 3e-4 in float32), accepting the small O(ξ) discontinuity at the switch.

Note: `mean` is unaffected (it uses the cancellation-free `expm1∘gammaln` form) — a template for option 1.

## References

- `src/xtremax/distributions/gevd.py` (`variance`, `skew`, `kurtosis`); same structure in `frechet.py` / `weibull.py` (reachable only at |ξ| ≥ their constrained ranges, so lower priority).
- Pinned loosely by `tests/test_oracles.py::TestClassOracles::test_moments` (asserts finiteness only inside the band, tight values elsewhere) — tighten that assertion when fixing.

## Relationships

- Related: #41, #51 (shared GUMBEL_THRESHOLD), PR #87 (oracle tests that found it)
